src/session/chat/mod.rs

```diff
@@ -556,4 +602,16 @@ impl Chat {
             Self::this_expression("title").upcast()
         }
     }
+
+    pub(crate) fn permissions(&self) -> BoxedChatPermissions {
+        self.imp().permissions.borrow().as_ref().unwrap().to_owned()
+    }
+
+    pub(crate) fn set_permissions(&self, permissions: BoxedChatPermissions) {
+        if self.imp().permissions.borrow().as_ref() == Some(&permissions) {
```

Comment:
```suggestion
        if self.permissions() == permissions {
```

Reply:
Hmm. I just saw that my change would crash, because you unwrap in `permissions()`. So either return an `Option` in `permissions` or ignore this. Depends on what @melix99 prefers.

Reply:
This is a bit tricky in the current state because all the properties are None by default. This will change with the new tdlib-gobject library, but for now I guess this is fine, this is also what we currently do in some other cases.